flytepropeller/pkg/compiler/validators/bindings.go

```diff
@@ -152,7 +164,8 @@ func validateBinding(w c.WorkflowBuilder, nodeID c.NodeID, nodeParam string, bin
 					return param.GetType(), []c.NodeID{val.Promise.NodeId}, true
 				}
 
-				errs.Collect(errors.NewMismatchingTypesErr(nodeID, val.Promise.Var, sourceType.String(), expectedType.String()))
+				errs.Collect(errors.NewMismatchingTypesErr(node.GetId(), outputVar, sourceType.String(), inputVar, expectedType.String()))
+				return nil, nil, !errs.HasErrors()
```

Comment:
is this right?

Reply:
yup, outputVar is a promise. https://github.com/flyteorg/flyte/blob/827d55b1b72cf6ca6051af13b3ff9dfc8f37a726/flytepropeller/pkg/compiler/validators/bindings.go#L126